src/Textarea/sgds-textarea.ts

```diff
@@ -0,0 +1,201 @@
+  defaultValue = '';
+
+  connectedCallback() {
+    super.connectedCallback();
+    this.resizeObserver = new ResizeObserver(() => this.setTextareaHeight());
+
+    this.updateComplete.then(() => {
+      this.setTextareaHeight();
+      this.resizeObserver.observe(this.textarea);
+    });
+  }
+
+  disconnectedCallback() {
+    super.disconnectedCallback();
+    this.resizeObserver.unobserve(this.textarea);
+  }
+
+  /** Sets focus on the textarea. */
+  focus(options?: FocusOptions) {
+    this.textarea.focus(options);
+  }
+
+  /** Checks for validity and shows the browser's validation message if the control is invalid. */
+  reportValidity() {
+    return this.textarea.reportValidity();
+  }
+
+  handleInvalid() {
+    this.invalid = true;
+  }
+
+  handleChange(event: string){
+    this.value = this.textarea.value;
+    this.emit(event);
+  }
+
+  handleFocus() {
+    this.hasFocus = true;
+    this.emit('sgds-focus');
+  }
+
+  handleBlur() {
+    this.hasFocus = false;
+    this.emit('sgds-blur');
+  }
+
+   /** Selects all the text in the textarea. */
+   select() {
+    this.textarea.select();
+  }
+
+  handleKeyDown(event: KeyboardEvent) {
+    const hasModifier = event.metaKey || event.ctrlKey || event.shiftKey || event.altKey;
+
+    // Pressing enter when focused on an input should submit the form like a native input, but we wait a tick before
+    // submitting to allow users to cancel the keydown event if they need to
+    if (event.key === 'Enter' && !hasModifier) {
+      setTimeout(() => {
+        if (!event.defaultPrevented) {
+          this.formSubmitController.submit();
+        }
+      });
+    }
+  }
+
+  @watch('rows', { waitUntilFirstUpdate: true })
+  handleRowsChange() {
+    this.setTextareaHeight();
+  }
+
+  setTextareaHeight() {
+    if (this.resize === 'auto') {
+      this.textarea.style.height = 'auto';
+      this.textarea.style.height = `${this.textarea.scrollHeight}px`;
+    } else {
+      (this.textarea.style.height as string | undefined) = undefined;
+    }
+  }
+
+  @watch('disabled', { waitUntilFirstUpdate: true })
+  handleDisabledChange() {
+    // Disabled form controls are always valid, so we need to recheck validity when the state changes
+    this.textarea.disabled = this.disabled;
+    this.invalid = !this.textarea.checkValidity();
+  }
+
+  @watch('value', { waitUntilFirstUpdate: true })
+  handleValueChange() {
+    this.invalid = !this.textarea.checkValidity();
+    this.valid = this.textarea.checkValidity()
+    this.updateComplete.then(() => this.setTextareaHeight());
+  }
+
+  render() {
+
+    // if maxlength is defined
+    const wordCount = html`
+    <div class="form-text">${this.value.length}/${this.maxlength}</div>
+    `
+
+    return html`
+
+        <div class="text-area-label-wrapper d-flex justify-content-between">
+          <label for=${ifDefined(this.textareaId)} class="form-label">${this.label}</label>
+          ${this.maxlength > "0" ? wordCount : undefined}
```

Comment:
if this.maxlength is a `string` type , you will actually not get the correct result( as string to string comparison evaluates by alphabetical order).  But luckily here this.maxlength is a Number so its evaluating like a number. 
see example:  https://www.w3schools.com/js/js_comparisons.asp#:~:text=When%20comparing%20a%20string%20with,NaN%20which%20is%20always%20false%20.&text=When%20comparing%20two%20strings%2C%20%222,1%20is%20less%20than%202.

This is related to qn above on whether maxlength and minlength is string or number

Safer to do number to number comparison. If maxlength/minlength is going to be a string type, then here you can convert the string to an integer like so.
` parseInt(this.maxlength) > 0 ? wordCount : undefined`

if maxlength/minlength is going to be a number type then even easier just, directly compare it with number 0  --> `this.maxlength > 0` rather than "0"

although comparing a number to string "0" will evaluate like numbers too because greater than /less than (>, <) operators are not strict comparisons.